src/test/java/server/EchoServerTest.java

```diff
@@ -1,12 +1,65 @@
 package server;
 
+
+import mocks.MockWrapper;
 import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
 import static org.junit.jupiter.api.Assertions.*;
 
 
-class EchoServerTest {
-    @Test void walkingSkeletonTest() {
-        EchoServer classUnderTest = new EchoServer();
-        assertEquals(classUnderTest.walkingSkeletonTest(), "testing 123");
+class MockWrappersTests {
+    @Test
+    public void testRunGetsAcceptsClient() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4000);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsToAcceptClient());
+    }
+    @Test
+    public void testRunCreatesInputStream() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4444);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsToCreateInputStream());
+    }
+    @Test
+    public void testRunCreatesOutputStream() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4001);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsToCreateOutStream());
+    }
+    @Test
+    public void testRunReceivesMessages() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4002);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsReceiveMessages());
     }
+//    @Test
```

Comment:
Please remove commented-out code.  You can stash this later, but think it's safe to say that, as a company, we pooh-pooh leaving any comments in committed code.

Reply:
removed comments